### PR TITLE
Form component improvements

### PR DESCRIPTION
**Pull request by tyom**
- Improve styling of select box (including fixes for IE9)
- Fix inline form-group styling for various use cases
- Remove top margin from form-group if it follows hidden input
- Allow multiple modifiers to be passed to form components (e.g. `modifier: 'inline'` and `modifier: ['inline', 'small']`)

### CODE REVIEW

File: assets/stylesheets/components/form/_form-control.scss [hunk 1]
```diff
@@ -48,6 +49,14 @@ select.c-form-control {
   &::-ms-expand {
     display: none;
   }
+
+  // IE9
+  @media all and (min-width:0\0) and (min-resolution:.001dpcm) {
```
Comment (ztolley) [inline assets/stylesheets/components/form/_form-control.scss]: Wow what the heck is that syntax?

File: src/templates/_macros/form.njk [hunk 2]
```diff
@@ -415,6 +426,7 @@
  # @param {string} [props.value] - Checked field value
  # @param {string} [props.children] - Main field subfields
  # @param {string} [props.options[].subfield] - Field options subfields
+ # @param {string} [props.initialOption] - Initial option label to use for empty value in select box
```
Comment (ztolley) [inline src/templates/_macros/form.njk]: 👍

File: assets/stylesheets/components/form/_form-control.scss (hunk 1 above)
Comment (jim68000) [inline assets/stylesheets/components/form/_form-control.scss]: min width is 0 plus one null byte?

Comment (tyom) [inline assets/stylesheets/components/form/_form-control.scss]: That hack is supposed to only target IE9. Saves creating a separate stylesheet for it. It's only been needed here so far.

**Pull request merged**